rest-api/pom.xml

```diff
@@ -56,6 +56,12 @@
             <version>${workflow.version}</version>
         </dependency>
         <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>pipeline-input-step</artifactId>
+            <version>2.0</version>
+            <optional>true</optional>
```

Comment:
> it's not clear to me how `optional` works here

For good reason—it is complicated.

In this case the Jenkins plugin manager should link `pipeline-rest-api` against `pipeline-input-step` if `pipeline-input-step` is there, otherwise it will just ignore the dependency.

What the update center UI will do is TBD; it is very hard to test that without having a release.